wrappers/src/results_cs.cpp

```diff
@@ -124,20 +123,18 @@ REALM_EXPORT Results* results_get_filtered_results(const Results& results, uint1
 {
     return handle_errors(ex, [&]() {
         Utf16StringAccessor query_string(query_buf, query_len);
-        auto query = results.get_query();
         auto const &realm = results.get_realm();
+        auto const& object_schema = results.get_object_schema();
 
-        parser::ParserResult result = parser::parse(query_string.to_string());
-
-        parser::KeyPathMapping mapping;
+        query_parser::KeyPathMapping mapping;
         realm::populate_keypath_mapping(mapping, *realm);
 
-        query_builder::NoArguments no_args;
-        query_builder::apply_predicate(query, result.predicate, no_args, mapping);
+        auto table = realm->read_group().get_table(object_schema.table_key);
 
-        DescriptorOrdering ordering;
-        query_builder::apply_ordering(ordering, query.get_table(), result.ordering);
-        return new Results(realm, std::move(query), std::move(ordering));
+        query_parser::NoArguments no_args;
+        auto query = table->query(query_string, no_args, mapping);
+        auto ordering = query.get_ordering();
+        return new Results(realm, query, *ordering);
```

Comment:
I had to brush up on move semantics. I don't think the `std::move` is needed in either of the case. One special rule says that if the move constructor and operator are defined, returning automatic l-values by copy from a method or function moves such values and not copy.
If my understanding is incorrect, then I'll put the `std::move` back where it was.